src/websocket/websockethandler.cpp

```diff
@@ -270,5 +293,8 @@ void WebSocketHandler::onMessageReceived(const QString& message) {
   logger.debug() << "Message received:" << message;
 
   PushMessage parsedMessage(message);
+  emit MozillaVPN::instance()->recordGleanEventWithExtraKeys(
+      GleanSample::pushMessageReceived, {{"type", parsedMessage.type()}});
```

Comment:
QVariant::fromValue(type).toString()

Reply:
Yeah, so this is what I had initially and it was reporting empty strings. That is why I implemented that whole method to make the enum into a string. Maybe I was doing it wrong though, that is quite plausible. I'll try again.